docs/txp.md

```diff
@@ -0,0 +1,434 @@
+    and adopted `BlockVersionData` (which contains data which slot is defined last for bootstrap era)
+
+#### Outputs checks
+
+For each output address we perform the following checks:
+1. If `verifyAllIsKnown` is `True`, address' attributes must be known.
+2. If `verifyAllIsKnown` is `True`, address' type must be known.
+3. Address can't be a redeem address.
+
+#### Inputs checks
+
+For each input `i` we perform:
+
+- *Inputs known check*: if `verifyAllIsKnown` is `True`, type of input `i` should be known
+- *UTXO check*: If input `i` is of type `TxInUtxo`, it should have corresponding record in UTXO
+- *Difference check*: input `i` must be different from all other inputs in transaction
+
+In addition we perform number of consolidated checks (checks which consider whole set of inputs):
+
+* Witness check (separately for each witness)
+* Sums check (check inputs sum is greater than or equal to outputs sum)
+* Transaction fee check (check transaction fee is fine)
+
+##### Witness checks
+
+Witness check for any type is two-fold:
+
+* check that witness corresponds to the address of corresponding input
+* check witness is correct
+
+We describe checks for each type in folowing sections.
+
+###### Public key witness
+
+Witness `PkWitness` contains:
+
+* public key `pk`
+* signature `sig`
+
+We require:
+
+* Address is a `PubKey` address
+* Address spending data is created with public key `pk` (\*)
+* Signature `sig` is a valid signature of transaction (validity checked with `pk`)
+
+Note: signature inside witness is given for `TxSigData` (which is basically the same as `Hash Tx`)
+  with `SignTx` tag.
+  
+Note (\*): even though address doesn't contain its spending data, it's easy
+to check whether given spending data corresponds to given
+address. Suppose we have an address `addr` and spending data
+`asd`. Let's say that `addr` has attributes `attrs` and type `t`. We
+can construct `Address'` from `attrs`, `t` and `asd` and compare its
+hash with the one from `addr`.
+  
+###### Script witness
+
+Witness `ScriptWitness` contains:
+  
+* validator script, `validator`
+* redeemer script, `redeemer`
+
+We require:
+  
+* Address is a `Script` address
+* Address spending data is created with validator script same as `validator`
+* Both `redeemer` and `validator` scripts have the same version.
+* If `verifyAllIsKnown` is `True`, script version (which is same for
+    redeemer and validator) must be known.
+    - For CSL 1.0 it means that version must be equal to 0, it's the only version known to our software.
+* Plutus validation script built from redeemer and validator for
+    given tx must return `True`
+
+###### Redeem witness
+
+Witness `PkWitness` contains:
+
+* redeem public key `pk`
+* redeem signature `sig`
+
+We require:
+
+
+* Address is a `Redeem` address 
+* Address spending data is created with redeem public key `pk`
+* Signature `sig` is a valid signature of transaction (validity checked with `pk`)
```

Comment:
**Important**: need an accurate description of how signature check is made:
 
- Which tag used
- What data included into tag

Reply:
Done.